astropy/coordinates/angles/core.py

```diff
@@ -403,36 +404,16 @@ def _wrap_at(self, wrap_angle):
         # Convert the wrap angle and 360 degrees to the native unit of
         # this Angle, then do all the math on raw Numpy arrays rather
         # than Quantity objects for speed.
-        a360 = u.degree.to(self.unit, 360.0)
-        wrap_angle = wrap_angle.to_value(self.unit)
         self_angle = self.view(np.ndarray)
-        if NUMPY_LT_2_0:
-            # Ensure ndim>=1 so that comparison is done using the angle dtype.
-            self_angle = self_angle[np.newaxis]
+        a360 = np.asarray(u.degree.to(self.unit, 360.0), dtype=self_angle.dtype)
```

Comment:
I'm not 100% sure that `np.asarray` would avoid a copy in scenarios where it's not needed. Is this any better ?
```suggestion
        a360 = u.degree.to_value(self.unit, 360.0).astype(self_angle.dtype, copy=COPY_IF_NEEDED)
```

Reply:
It is the recommended way as of the numpy 2.0 migration guide: https://numpy.org/devdocs/numpy_2_0_migration_guide.html#adapting-to-changes-in-the-copy-keyword

Reply:
the `asarray` is needed to make it something that has `astype` in the first place.

Also, units don't have `to_value`.

Reply:
```

In [7]: u.degree.to(u.rad, 360.0)
Out[7]: 6.283185307179586

In [8]: u.degree.to(u.rad, 360.0).astype(np.float32)
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[8], line 1
----> 1 u.degree.to(u.rad, 360.0).astype(np.float32)

AttributeError: 'float' object has no attribute 'astype'

```

Reply:
Also: this will *always* copy since we will convert a python float to the wished dtype

Reply:
oh my, 3 mistakes in one line of suggestion, this must be my new record 🎉 
(seriously though, thanks for checking, and sorry I wasn't careful enough here)